src/dagmc.cpp

```diff
@@ -20,28 +38,100 @@ moab::DagMC* DAG;
 
 } // namespace model
 
+
+bool get_uwuw_materials_xml(std::string& s) {
+  UWUW uwuw(DAGMC_FILENAME.c_str());
+
+  std::stringstream ss;
+  bool uwuw_mats_present = false;
+  if (uwuw.material_library.size() != 0) {
+    uwuw_mats_present = true;
+    // write header
+    ss << "<?xml version=\"1.0\"?>\n";
+    ss << "<materials>\n";
+    std::map<std::string, pyne::Material> mat_lib = uwuw.material_library;
+    // write materials
+    for (auto mat : mat_lib) { ss << mat.second.openmc("atom"); }
+    // write footer
+    ss << "</materials>";
+    s = ss.str();
+  }
+
+  return uwuw_mats_present;
+}
+
+pugi::xml_document* read_uwuw_materials() {
+  pugi::xml_document* doc = NULL;
+
+  std::string s;
+  bool found_uwuw_mats = get_uwuw_materials_xml(s);
+  if (found_uwuw_mats) {
+    doc = new pugi::xml_document();
+    pugi::xml_parse_result result = doc->load_string(s.c_str());
+  }
+  return doc;
+}
+
+bool write_uwuw_materials_xml() {
+  std::string s;
+  bool found_uwuw_mats = get_uwuw_materials_xml(s);
+    // if there is a material library in the file
+  if (found_uwuw_mats) {
+    // write a material.xml file
+    std::ofstream mats_xml("materials.xml");
+    mats_xml << s;
+    mats_xml.close();
+  }
+
+  return found_uwuw_mats;
+}
+
 void load_dagmc_geometry()
 {
   if (!model::DAG) {
     model::DAG = new moab::DagMC();
   }
 
-  int32_t dagmc_univ_id = 0; // universe is always 0 for DAGMC
+  /// Materials \\\
+
+  // create uwuw instance
+  UWUW uwuw(DAGMC_FILENAME.c_str());
+
+  // check for uwuw material definitions
+  bool using_uwuw = (uwuw.material_library.size() == 0) ? false : true;
```

Comment:
`bool using_uwuw = (uwuw.material_library.size() > 0);`
or
`bool using_uwuw = !uwuw.material_library.empty();`

Reply:
Eesh that was silly. Yeah these are better. Thanks!